cmd/contour/contour.go

```diff
@@ -16,6 +16,9 @@ package main
 import (
 	"os"
 
+	// Automatically set GOMAXPROCS to match the number of CPUs available to the container
+	_ "go.uber.org/automaxprocs"
```

Comment:
i think maybe instead of implementing it this way with a side effect import, let's be more explicit about using the package so we don't accidentally undo the behavior we introduce in the future etc.

maybe see if we can write our own init function or something that hooks up our logger to the `maxprocs .Set` function

Reply:
That makes sense @sunjayBhatia. I'll rework it to use `maxprocs.Set` so it is explicit and we get logging.

Reply:
The uber library is relatively well known as intentionally been created to set `GOMAXPROCS` with as little fuss as possible. 

Would recommend reusing the code as is vs writing a custom implementation.

Reply:
the suggestion here isn't to rewrite the library, but to directly use the helper function rather than the side-effect import so we have more visibility and don't accidentally undo any changes here, get logging from the library, etc.

Reply:
Ok if I add it in?

Reply:
Created https://github.com/projectcontour/contour/pull/5211 as this PR ran into quite a few conflicts due to age.